geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -283,4 +292,77 @@ public String toString() {
   public KnownVersion[] getSerializationVersions() {
     return null;
   }
+
+  private void validateScoreIsDouble(byte[] score) {
+    String scoreString = Coder.bytesToString(score).toLowerCase();
+    switch (scoreString) {
+      case "inf":
+      case "infinity":
+      case "+inf":
+      case "+infinity":
+      case "-inf":
+      case "-infinity":
+        return;
+      default:
+        if (!Coder.bytesToString(score).matches("^[-+]?[0-9]*\\.?[0-9]+([eE][-+]?[0-9]+)?$")) {
+          throw new NumberFormatException(ERROR_NOT_A_VALID_FLOAT);
+        }
+    }
+  }
+
+  static class OrderedSetEntry implements Comparable<OrderedSetEntry> {
+    public byte[] member;
+    public Double score;
```

Comment:
Should this be a double (lowercase, primitive)?

Reply:
These members can be private and final I think.

Reply:
Good call on the private and final. However, when sorting we use Double's compareTo(), which is not available on the double primitive.